catalog/app/containers/Bucket/Queries/Athena/Athena.tsx

```diff
@@ -1,458 +1,212 @@
-  return (error: Error) => <Alert error={error} title={title} />
-}
-
-const useStyles = M.makeStyles((t) => ({
-  emptySelect: {
-    margin: t.spacing(5.5, 0, 0),
-  },
-  form: {
-    margin: t.spacing(0, 0, 4),
-  },
-  results: {
-    margin: t.spacing(4, 0, 0),
-  },
-  sectionHeader: {
+const useResultsContainerStyles = M.makeStyles((t) => ({
+  breadcrumbs: {
     margin: t.spacing(0, 0, 1),
   },
-  select: {
-    flexBasis: '40%',
-    '& + &': {
-      flexBasis: '60%',
-      marginLeft: t.spacing(3),
-      marginBottom: t.spacing(-3), // counterpart for Select's optional description
-    },
-  },
-  selects: {
-    display: 'flex',
-    margin: t.spacing(3, 0),
-  },
 }))
 
-const useFormStyles = M.makeStyles((t) => ({
-  actions: {
-    margin: t.spacing(2, 0),
-  },
-  error: {
-    margin: t.spacing(1, 0, 0),
-  },
-  viewer: {
-    margin: t.spacing(3, 0, 0),
-  },
-}))
-
-interface FormProps {
-  disabled: boolean
-  error?: Error
-  onChange: (value: string) => void
-  onSubmit: (value: string) => () => void
-  value: string | null
+interface ResultsContainerProps {
+  bucket: string
+  className: string
+  queryExecutionId: string
+  results: QueryResults
+  workgroup: requests.athena.Workgroup
 }
 
-function Form({ disabled, error, value, onChange, onSubmit }: FormProps) {
-  const classes = useFormStyles()
-
-  const handleSubmit = React.useMemo(() => onSubmit(value || ''), [onSubmit, value])
-
+function ResultsContainer({
+  bucket,
+  className,
+  queryExecutionId,
+  results,
+  workgroup,
+}: ResultsContainerProps) {
+  const classes = useResultsContainerStyles()
   return (
-    <div>
-      <QueryEditor className={classes.viewer} onChange={onChange} query={value || ''} />
-
-      {error && (
-        <Lab.Alert className={classes.error} severity="error">
-          {error.message}
-        </Lab.Alert>
-      )}
-
-      <div className={classes.actions}>
-        <M.Button
-          variant="contained"
-          color="primary"
-          disabled={disabled}
-          onClick={handleSubmit}
-        >
-          Run query
-        </M.Button>
-      </div>
+    <div className={className}>
+      <ResultsBreadcrumbs
+        bucket={bucket}
+        className={classes.breadcrumbs}
+        queryExecutionId={queryExecutionId}
+        workgroup={workgroup}
+      />
+      {results.data.case({
+        Init: () => null,
+        Ok: (queryResults) => {
+          if (queryResults.rows.length) {
+            return (
+              <Results
+                rows={queryResults.rows}
+                columns={queryResults.columns}
+                onLoadMore={
+                  queryResults.next ? () => results.loadMore(queryResults) : undefined
+                }
+              />
+            )
+          }
+          if (queryResults.queryExecution) {
+            return (
+              <History
+                bucket={bucket}
+                executions={[queryResults.queryExecution!]}
```

Comment:
`!` doesn't seem necessary here, bc `if (queryResults.queryExecution)` check asserts it's truthy